seonshine_frontend/src/pages/menuManagement/components/ModalMenuItem.tsx

```diff
@@ -0,0 +1,331 @@
+        queryClient.invalidateQueries({ queryKey: ['getMenuList'] });
+        toast.success(res.message);
+        setIsEditing(false);
+        handleCloseModal();
+      },
+      onError: () => {
+        toast.error('Update restaurant failed!');
+      },
+    });
+  };
+
+  const handleCancel = () => {
+    setIsEditing(false);
+    reset({
+      name: selectedItem?.name || '',
+    });
+  };
+
+  const handleClickAction = () => {
+    setIsConfirmModalOpen(true);
+  };
+
+  const handleDeleteItem = () => {
+    deleteMenuItem(undefined, {
+      onSuccess: (data) => {
+        toast.success(data.message);
+        setIsModalOpen(false);
+        queryClient.invalidateQueries({ queryKey: ['getMenuList'] });
+      },
+      onError: () => {
+        toast.error('Delete menu item failed!');
+      },
+    });
+  };
+
+  const handleCloseModal = () => {
+    setIsModalOpen(false);
+    setIsEditing(false);
+    reset();
+  };
+
+  const handleCreate = (data: { name: string }) => {
+    const payload: CreateMenuItemPayloadType = {
+      ...data,
+      file: imageFile,
+      restaurant_id: restaurant_id,
+    };
+
+    createMenuItem(payload, {
+      onSuccess: (response) => {
+        toast.success(response.message);
+        queryClient.invalidateQueries({ queryKey: ['getMenuList'] });
+        reset();
+        setImageFile(null);
+        setIsModalOpen(false);
+      },
+      onError: () => {
+        toast.error('Create menu item failed!');
+      },
+    });
+  };
+
+  const handleEditToggle = () => setIsEditing(!isEditing);
+
+  const handleImageChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
+    setUploadError(null);
+
+    if (event.target.files && event.target.files.length > 0) {
+      const file = event.target.files[0];
+      const validationResult = isValidImageFile(file);
+
+      if (!validationResult.isValid) {
+        setUploadError(validationResult.messageError || null);
+        return;
+      }
+
+      setImageFile(file);
+
+      const reader = new FileReader();
+      reader.onload = () => {
+        setImageUrl(reader.result as string);
+      };
+      reader.readAsDataURL(file);
+    }
+  };
+
+  useEffect(() => {
+    if (selectedItem) {
+      reset({
+        name: selectedItem.name,
+      });
+    }
+  }, [selectedItem]);
+
+  return (
+    <Modal
+      open={isOpen}
+      onClose={handleCloseModal}
+    >
+      <Box className="absolute top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2 w-3/5 md:w-1/3 lg:w-1/5 bg-white shadow-xl rounded-lg outline-none">
+        <Box className="h-full flex flex-col justify-between">
+          <Stack
+            className="rounded-md p-8 box-border cursor-pointer bg-gray-100 flex flex-col items-center"
+            direction="column"
+            gap={2}
+          >
+            <Box className="w-full h-48 md:h-64 bg-gray-200 flex items-center justify-center overflow-hidden rounded-md">
+              {isCreateModal && !imageUrl ? (
```

Comment:
with display image section. We just check imageUrl exist or not. Define one imageUrl for combine selectedItem?.image_url and URL from upload local